Plots/WRF/NCL_WRF_zoom_1_2.py

```diff
@@ -0,0 +1,128 @@
+"""
+NCL_WRF_zoom_1_2.py
+==================
+This script illustrates the following concepts:
+    - Plotting WRF data on native grid
+    - Subsetting data to 'zoom in' on an area
+    - Plotting data using wrf python functions
+    - Following best practices when choosing a colormap
+    
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/wrf_zoom_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/wrf_zoom_1_2_lg.png
+"""
+
+###############################################################################
+# Import packages
+
+from netCDF4 import Dataset
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mticker
+import cartopy.crs as ccrs
+from cartopy.feature import NaturalEarthFeature
+import cartopy.feature as cfeature
+
+from wrf import (getvar, to_np, latlon_coords, get_cartopy)
+import geocat.datafiles as gdf
+
+###############################################################################
+# Read in the data
+
+wrfin = Dataset(gdf.get("netcdf_files/wrfout_d03_2012-04-22_23_00_00_subset.nc"))
+
+td2 = getvar(wrfin, "td2")
+
+###############################################################################
+# Create a subset of the data for zoomed in projection
+
+dims = td2.shape
+
+y_start = int(dims[0]/2)
+y_end = int(dims[0]-1)
+
+x_start = int(0)
+x_end = int(dims[1]/2)
+
+td2_zoom = td2[y_start:y_end, x_start:x_end]
+
+# Define the latitude and longitude coordinates
+lats, lons = latlon_coords(td2_zoom)
+
+###############################################################################
+# Plot the data
+
+# The `get_cartopy` wrf function will automatically find and use the 
+# intended map projection for this dataset 
+cart_proj = get_cartopy(td2_zoom)
+
+fig = plt.figure(figsize=(12,12))
+ax = plt.axes(projection=cart_proj)
+
+# Add features to the projection
+states = NaturalEarthFeature(category="cultural",
+                             scale="50m",
+                             facecolor="none",
+                             name="admin_1_states_provinces_shp")
+
+ax.add_feature(cfeature.LAKES, edgecolor='lightgray', facecolor='gold')
```

Comment:
Is this line needed? I don't see any lakes drawn on the original plot.

Reply:
This line came from the outline that is around Lake Tahoe, which would imply some call to adding lakes to the projection. After you pointed this out, I double checked the NCL code and original NCL projection to see if there was any mention of lakes. There was not, so I wanted to double check that maybe the color is representative to the temperature over the lake, not just a geographic overlay. It appears this is what the reasoning is, but I still don't know why Tahoe is outlined and no other lakes are. That being said, I did also run the projection with the default NCL color scheme to double check consistency with the Python version, so I am attaching that here for your reference as well (but the color scheme will remain magma in the code). In the mean time, I will go ahead and take out the line for adding lake outlines/fills!

![image](https://user-images.githubusercontent.com/43652875/95120813-552d5100-070b-11eb-9340-109fbe95864c.png)

Reply:
Sounds good thank you!